src/Platform/Microsoft.Testing.Platform/TestHostControllers/ITestHostControllerInfo.cs

```diff
@@ -7,5 +7,11 @@ internal interface ITestHostControllerInfo
 {
     bool HasTestHostController { get; }
 
+    /// <summary>
+    /// Gets information whether the current process is a test controller or not.
+    /// When null the value has not been set yet, it is figured out while we are building the test application.
+    /// </summary>
+    bool? CurrentProcessIsTestHostController { get; }
```

Comment:
I like the design of having boolean properties prefixed with `Is`, `Can`, `Has`, maybe here we should go with `IsCurrentProcessTestHostController`?

Reply:
Renamed.